Filters/EmbedUrlFilter.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Fluid;
+using Fluid.Values;
+using OrchardCore.Liquid;
+using System.Threading.Tasks;
+
+namespace Etch.OrchardCore.Widgets.Filters
+{
+    public class EmbedUrlFilter : ILiquidFilter
+    {
+        public ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, LiquidTemplateContext context)
+        {
+            var videoUrl = input.ToObjectValue().ToString();
+
+            if (videoUrl.Contains("youtube.com/watch?v="))
+            {
+                return new ValueTask<FluidValue>(new StringValue(ConvertToEmbed(videoUrl)));
+            }
+
+            return new ValueTask<FluidValue>(new StringValue(videoUrl));
+        }
+
+        public static string ConvertToEmbed(string url)
+        {
+            url = url.Replace("watch?v=", "embed/");
+
+            if (url.IndexOf("&") >= 0)

```

Comment:
i removed the check .Contains("&"), IndexOf() can both check and return the index if it exists right?